File: hermes/src/main/java/com/lodz/android/hermes/contract/OnSubscribeListener.java
package com.lodz.android.hermes.contract;

/**
 * 推送接口回调
 * Created by zhouL on 2018/5/23.
 */
public interface OnSubscribeListener {

    /**
     * 主题订阅成功
     * @param topic 主题名称
     */
    void onSubscribeSuccess(String topic);

    /**
     * 主题订阅失败
     * @param topic 主题名称
     * @param cause 异常
     */
    void onSubscribeFailure(String topic, Throwable cause);

    /**
     * 后台消息到达
     * @param subTopic 订阅主题
     * @param msg 消息
     */
    void onMsgArrived(String subTopic, String msg);
}

File: hermes/src/main/java/com/lodz/android/hermes/mqtt/OnSubscribeListener.kt
package com.lodz.android.hermes.mqtt

/**
 * 推送接口回调
 * @author zhouL
 * @date 2019/12/20
 */
interface OnSubscribeListener {

    /** 主题订阅成功，主题名称[topic] */
    fun onSubscribeSuccess(topic: Array<String>)

    /** 主题订阅失败，主题名称[topic]，异常[cause] */
    fun onSubscribeFailure(topic: Array<String>, cause: Throwable)

    /** 后台消息到达，订阅主题[subTopic]，消息[msg] */
    fun onMsgArrived(subTopic: String, msg: String)
}